File: python/62a_1552A_SubsequencePermutation.py
def solve(s, n):
    cnt = 0
    temp = list(s)
    temp.sort()
    for i in range(n):
        if s[i] != temp[i]:
            cnt += 1
    return cnt

def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        s = input()
        ans.append(solve(s, n))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/62a_1552A_SubsequencePermutation.cpp
#include<bits/stdc++.h>

using namespace std;

int solve(string s, int n){
    int cnt = 0;
    string temp = s;
    sort(temp.begin(), temp.end());
    for(int i = 0; i < n; i++){
        if(s[i] != temp[i]){
            cnt++;
        }
    }
    return cnt;
}

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);

    int t, n;
    string s;
    vector<int> ans;
    cin>>t;
    for(int i = 0; i < t; i++){
        cin>>n;
        cin>>s;
        ans.push_back(solve(s, n));
    }

    for(auto i : ans){
        cout<<i<<"\n";
    }
    return 0;
}